cli/cmd/check_endpoint_status.go

```diff
@@ -0,0 +1,38 @@
+package cmd
```

Comment:
Generally speaking, the implementation looks great!

However, I'm wondering if implementing this functionality as a command is the right way to go. I think this should either live in `cli/pkg`.

Reply:
I have no strong opinion in which package this method is implemented:
- It fits in `cmd` as we only use it for commands
- It also fits in the `cli/pkg` because it is some kind of utility.

Reply:
> I had some doubts regarding the SonarCloud tests failing. So there are 4 files in which a similar kind of error handling is done and hence code duplication is being shown. So should I try to clear that up or its okay?

The code duplication that's listed is something that I am not considering a problem at all ;) (I mean, you're literally catching an error and printing it to console, it's not like the code gets better or more readable if you refactor it).

Anyway, our sonarcloud integration is configured to be more of an advisory, not a compulsory check. so you can leave it as it is :)